### ISSUE TITLE
`cargo clippy` fails with `Unrecognized option: 'allow-dirty'`

### ISSUE THREAD

**Issue by jon-chuang** [rust-lang/rust-clippy#13713]
### Summary

```
cargo clippy --fix --allow-dirty --allow-staged -- -D warnings
warning: virtual workspace defaulting to `resolver = "1"` despite one or more workspace members being on edition 2021 which implies `resolver = "2"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
note: for more details see https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions
error: process didn't exit successfully: `/Users/Jon-Chuang/.rustup/toolchains/stable-aarch64-apple-darwin/bin/clippy-driver /Users/Jon-Chuang/.rustup/toolchains/stable-aarch64-apple-darwin/bin/rustc -vV` (exit status: 1)
--- stderr
error: Unrecognized option: 'allow-dirty'
```

rustc:
```
 rustc 1.82.0 (f6e511eec 2024-10-15)
```

### Version

```text
rustc 1.82.0 (f6e511eec 2024-10-15)
binary: rustc
commit-hash: f6e511eec7342f59a25f7c0534f1dbea00d01b14
commit-date: 2024-10-15
host: aarch64-apple-darwin
release: 1.82.0
LLVM version: 19.1.1
```

**Comment by jon-chuang** [comment]
Weird, went away once I ran a build. Still, I think this is a bug.

**Comment by Alexendoo** [comment]
Should be fixed in 1.83 by https://github.com/rust-lang/rust-clippy/pull/13468
